torch/csrc/distributed/c10d/symm_mem/NCCLSymmetricMemory.cu

```diff
@@ -65,29 +66,17 @@ static __global__ void build_ptr_dev(
 
 class NCCLSymmetricMemory : public SymmetricMemory {
  public:
-#ifdef NCCL_HAS_SYMMEM_DEVICE_SUPPORT
- NCCLSymmetricMemory(
-      std::shared_ptr<NCCLAllocation> allocation,
-      const std::string& group_name,
-      ncclWindow_t buffer_handle,
-      ncclWindow_t signal_handle,
-      ncclDevComm devComm)
-#else
  NCCLSymmetricMemory(
       std::shared_ptr<NCCLAllocation> allocation,
       const std::string& group_name,
       ncclWindow_t buffer_handle,
       ncclWindow_t signal_handle)
-#endif
       : allocation_(allocation),
```

Comment:
Also allocation should be moved here, and all other member initializers should reference allocation_